randomUsernameAndPassword/assignment1/Q2.py

```diff
@@ -0,0 +1,54 @@
+import unittest
+
+class Node:
+  def __init__(self, values):
+    assert(values)
+    head, tail = values[0], values[1:]
+    self.data = head
+    if tail:
+      self.next = Node(tail)
+    else:
+      self.next = None
+
+  def get_next(self):
+    return self.next
+
+  def get_data(self):
+    return self.data
+
+def get_kth_last(head, k):
```

Comment:
Do you think you can get the bracket notation to work?

list = Node([4, 2, 5, 6, 9, 8])
list[-2] -> 6

Bracket notation for ordered container is more common, it is easier for your users who do not have to learn a new API for a new container.

To make the bracket notation work, you need to implement __getitem__ on the Node class.

Reply:
Really nice improvement, thank you!